mantidimaging/gui/windows/main/presenter.py

```diff
@@ -116,6 +116,14 @@ def create_new_stack(self, container: Union[Images, Dataset], title: str):
             if container.sample.has_proj180deg() and container.sample.proj180deg.filenames:
                 self._add_stack(container.sample.proj180deg, container.sample.proj180deg.filenames[0], sample_dock)
 
+        if len(current_stack_visualisers) > 1:
+            tab_bar = self.view.findChild(QTabBar)
+            if tab_bar is not None:
+                last_stack_pos = len(current_stack_visualisers) - 1
+                # make Qt process the addition of the dock onto the main window
+                QApplication.processEvents()
```

Comment:
I wonder if QApplication.sendPostedEvents() will do what we want here?

Reply:
👍 Works with that